service/service.go

```diff
@@ -222,8 +222,7 @@ func InstallAndStart(svc ServiceActions) error {
 		if err != nil {
 			logger.Errorf("retrying start request (%v)", errors.Cause(err))
 		}
-
-		if err = svc.Start(); err == nil {
+		if err = restartOrStart(svc); err == nil {
```

Comment:
Can you please add a comment here as to why we're *restarting*, as it's not obvious from the context. From your PR, I gather that it's because svc.Install may regenerate the service file in a different way.